### DISCUSSION TITLE
JSN-SR04T V3.0 shows wrong distance - HC-SR04 is working fine on Wemos D1 mini

### DISCUSSION THREAD

**Discussion by FillyderErste** [arendst/Tasmota#18966] [Q&A]
Hi everyone,

I want to replace a HC-SR04 by a [JSN-SR04T V3.0 ](https://protosupplies.com/product/jsn-sr04t-v3-0-waterproof-ultrasonic-range-finder/) connected to a Wemos D1 mini.
The HC-SR04 is working fine, so I think my configuration is OK. But when I connect the JSN-SR04T to the Wemos D1 mini, the measured values are not reasonable anymore. Acc. to [tasmota](https://tasmota.github.io/docs/HC-SR04/) no change of the configuration should be necessary, at least this is my understanding.
With the JSN-SR04T connected, the log shows "CRC error" and "Mode 1" is started. Is this behaviour OK?
I tried 2 different JSN-SR04Ts, but the result is the same.

Also I tried to use a different working mode of the JSN-SR04T as described here https://github.com/arendst/Tasmota/discussions/15590, but Tasmota did not detect the different working mode.

Did anyone has the same issue and can give me some advice? Since I could not find an comparable issue, discussison.

**Log startup of HC-SR04**
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F4, Count 120
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 448/699, Busy 0
00:00:00.071 ROT: Mode 1
00:00:00.074 SRC: Restart
00:00:00.076 Project tasmota - Tasmota water level Version 12.5.0(sensors)-2_7_4_9(2023-04-17T08:03:55)
00:00:00.180 WIF: Checking connection...
00:00:00.181 WIF: Attempting connection...
00:00:00.518 WIF: Connecting to AP1 in mode 11n as tasmota-0C1932-6450...
00:00:01.471 SR4: Received '00'
00:00:01.486 SR4: Received '00'
00:00:01.500 SR4: Received '00'
00:00:01.718 SR4: Mode 1
00:00:02.005 WIF: Checking connection...
00:00:02.006 WIF: Connected
00:00:02.261 HTP: Web server active on tasmota-0C1932-6450 with IP address 192.168.178.73
00:00:02.434 WIF: Sending Gratuitous ARP
00:00:02.435 NTP: Sync time...
00:00:02.459 WIF: DNS resolved '2.pool.ntp.org' (78.47.168.188) in 23 ms
00:00:02.503 RTC: UTC 2023-06-25T16:03:52, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
17:03:52.000 RTC: Synced by NTP
17:03:52.285 MQT: Attempting connection...
17:03:52.287 WIF: DNS resolved '192.168.178.59' (192.168.178.59) in 1 ms
17:03:52.302 MQT: Connected
17:03:52.305 MQT: tele/tasmota_0C1932/LWT = Online (retained)
17:03:52.308 MQT: cmnd/tasmota_0C1932/POWER = 
17:03:52.310 MQT: Subscribe to cmnd/tasmota_0C1932/#
17:03:52.312 MQT: Subscribe to cmnd/tasmotas/#
17:03:52.314 MQT: Subscribe to cmnd/Wasserstandssensor_fb/#
17:03:52.322 MQT: tele/tasmota_0C1932/INFO1 = {"Info1":{"Module":"Generic","Version":"12.5.0(sensors)","FallbackTopic":"cmnd/Wasserstandssensor_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:03:52.338 MQT: tele/tasmota_0C1932/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-0C1932-6450","IPAddress":"192.168.178.73"}}
17:03:52.350 MQT: tele/tasmota_0C1932/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":72}}
17:03:55.489 QPC: Reset
17:03:55.625 HTP: Main Menu
17:03:56.417 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:03:56","Uptime":"0T00:00:08","UptimeSec":8,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":33,"MqttCount":1,"Wifi":{"AP":1,"SSId":"","Channel":6,"Mode":"11n","RSSI":86,"Signal":-57,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:03:56.445 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:03:56","SR04":{"Distance":58.7}}
17:03:57.130 HTP: Console
17:03:57.435 APP: Boot Count 72
17:03:57.819 CFG: Saved to flash at FB, Count 121, Bytes 4096
17:04:01.468 MQT: tasmota/discovery/4C75250C1932/config = {"ip":"192.168.178.73","dn":"Tasmota water level","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-0C1932-6450","mac":"4C75250C1932","md":"Generic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"12.5.0","t":"tasmota_0C1932","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"sht":[[0,0,0],[0,0,0],[0,0,0],[0,0,0]],"ver":1} (retained)
17:04:01.547 MQT: tasmota/discovery/4C75250C1932/sensors = {"sn":{"Time":"2023-06-25T17:04:01","SR04":{"Distance":65.9}},"ver":1} (retained)

**Log startup of JSN-SR04T**
`JSN-SR04T-V3.0 - Mode 0 (default)
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F5, Count 119
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 448/699, Busy 0
00:00:00.071 ROT: Mode 1
00:00:00.074 SRC: Restart
00:00:00.076 Project tasmota - Tasmota water level Version 12.5.0(sensors)-2_7_4_9(2023-04-17T08:03:55)
00:00:00.180 WIF: Checking connection...
00:00:00.181 WIF: Attempting connection...
00:00:00.518 WIF: Connecting to AP1 in mode 11n as tasmota-0C1932-6450...
00:00:01.422 SR4: Received 'FEFC00FE'
00:00:01.423 SR4: CRC error
00:00:01.443 SR4: Received '00FE00E0'
00:00:01.444 SR4: CRC error
00:00:01.465 SR4: Received '0000E0E0'
00:00:01.681 SR4: Mode 1
00:00:02.028 WIF: Checking connection...
00:00:02.029 WIF: Connected
00:00:02.283 HTP: Web server active on tasmota-0C1932-6450 with IP address 192.168.178.73
00:00:02.461 WIF: Sending Gratuitous ARP
00:00:02.462 NTP: Sync time...
00:00:02.481 WIF: DNS resolved '2.pool.ntp.org' (173.249.33.207) in 17 ms
00:00:02.514 RTC: UTC 2023-06-25T16:01:51, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
17:01:51.001 RTC: Synced by NTP
17:01:52.292 MQT: Attempting connection...
17:01:52.294 WIF: DNS resolved '192.168.178.59' (192.168.178.59) in 0 ms
17:01:52.314 MQT: Connected
17:01:52.317 MQT: tele/tasmota_0C1932/LWT = Online (retained)
17:01:52.319 MQT: cmnd/tasmota_0C1932/POWER = 
17:01:52.321 MQT: Subscribe to cmnd/tasmota_0C1932/#
17:01:52.323 MQT: Subscribe to cmnd/tasmotas/#
17:01:52.326 MQT: Subscribe to cmnd/Wasserstandssensor_fb/#
17:01:52.334 MQT: tele/tasmota_0C1932/INFO1 = {"Info1":{"Module":"Generic","Version":"12.5.0(sensors)","FallbackTopic":"cmnd/Wasserstandssensor_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:01:52.348 MQT: tele/tasmota_0C1932/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-0C1932-6450","IPAddress":"192.168.178.73"}}
17:01:52.361 MQT: tele/tasmota_0C1932/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":71}}
17:01:52.579 HTP: Main Menu
17:01:53.939 HTP: Console
17:01:55.493 QPC: Reset
17:01:56.433 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:01:56","Uptime":"0T00:00:09","UptimeSec":9,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":24,"MqttCount":1,"Wifi":{"AP":1,"SSId"","Channel":6,"Mode":"11n","RSSI":90,"Signal":-55,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:01:56.457 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:01:56","SR04":{"Distance":24.8}}
17:01:57.441 APP: Boot Count 71
17:01:57.856 CFG: Saved to flash at F4, Count 120, Bytes 4096
17:02:01.471 MQT: tasmota/discovery/4C75250C1932/config = {"ip":"192.168.178.73","dn":"Tasmota water level","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-0C1932-6450","mac":"4C75250C1932","md":"Generic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"12.5.0","t":"tasmota_0C1932","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"sht":[[0,0,0],[0,0,0],[0,0,0],[0,0,0]],"ver":1} (retained)
17:02:01.553 MQT: tasmota/discovery/4C75250C1932/sensors = {"sn":{"Time":"2023-06-25T17:02:01","SR04":{"Distance":0.3}},"ver":1} (retained)
17:02:06.440 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:02:06","Uptime":"0T00:00:19","UptimeSec":19,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":22,"MqttCount":1,"Wifi":{"AP":1,"SSId"","Channel":6,"Mode":"11n","RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:02:06.468 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:02:06","SR04":{"Distance":7.1}}
17:02:11.036 WIF: Checking connection...
17:02:16.422 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:02:16","Uptime":"0T00:00:29","UptimeSec":29,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId","Channel":6,"Mode":"11n","RSSI":90,"Signal":-55,"LinkCount":1,"Downtime":"0T00:00:03"}}
17:02:16.451 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:02:16","SR04":{"Distance":0.2}}

**Log startup of JSN-SR04T M2-Pads shorted**
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F9, Count 123
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 448/699, Busy 0
00:00:00.071 ROT: Mode 1
00:00:00.073 SRC: Restart
00:00:00.076 Project tasmota - Tasmota water level Version 12.5.0(sensors)-2_7_4_9(2023-04-17T08:03:55)
00:00:00.180 WIF: Checking connection...
00:00:00.181 WIF: Attempting connection...
00:00:00.517 WIF: Connecting to AP1 in mode 11n as tasmota-0C1932-6450...
00:00:01.559 SR4: Received ''
00:00:01.661 SR4: Received ''
00:00:01.762 SR4: Received ''
00:00:01.983 SR4: Mode 1
00:00:02.630 WIF: Checking connection...
00:00:02.631 WIF: Attempting connection...
00:00:03.610 WIF: Checking connection...
00:00:03.611 WIF: Connected
00:00:03.765 HTP: Web server active on tasmota-0C1932-6450 with IP address 192.168.178.73
00:00:04.446 WIF: Sending Gratuitous ARP
00:00:04.447 NTP: Sync time...
00:00:04.499 WIF: DNS resolved '2.pool.ntp.org' (185.244.195.159) in 51 ms
00:00:04.533 RTC: UTC 2023-06-25T16:57:51, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
17:57:51.001 RTC: Synced by NTP
17:57:51.221 MQT: Attempting connection...
17:57:51.222 WIF: DNS resolved '192.168.178.59' (192.168.178.59) in 0 ms
17:57:51.239 MQT: Connected
17:57:51.242 MQT: tele/tasmota_0C1932/LWT = Online (retained)
17:57:51.244 MQT: cmnd/tasmota_0C1932/POWER = 
17:57:51.246 MQT: Subscribe to cmnd/tasmota_0C1932/#
17:57:51.248 MQT: Subscribe to cmnd/tasmotas/#
17:57:51.249 MQT: Subscribe to cmnd/Wasserstandssensor_fb/#
17:57:51.253 MQT: tele/tasmota_0C1932/INFO1 = {"Info1":{"Module":"Generic","Version":"12.5.0(sensors)","FallbackTopic":"cmnd/Wasserstandssensor_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:57:51.256 MQT: tele/tasmota_0C1932/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-0C1932-6450","IPAddress":"192.168.178.73"}}
17:57:51.258 MQT: tele/tasmota_0C1932/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":74}}
17:57:52.488 QPC: Reset
17:57:53.748 HTP: Main Menu
17:57:54.422 APP: Boot Count 74
17:57:55.325 CFG: Saved to flash at F8, Count 124, Bytes 4096
17:57:55.441 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:57:55","Uptime":"0T00:00:10","UptimeSec":10,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":31,"MqttCount":1,"Wifi":{"AP":1,"SSId:"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:05"}}
17:57:55.449 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:57:55","SR04":{"Distance":0.0}}
17:57:56.833 HTP: Console
17:58:00.438 MQT: tasmota/discovery/4C75250C1932/config = {"ip":"192.168.178.73","dn":"Tasmota water level","fn":["Tasmota",null,null,null,null,null,null,null],"hn":"tasmota-0C1932-6450","mac":"4C75250C1932","md":"Generic","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"12.5.0","t":"tasmota_0C1932","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":0,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[0,0,0,0],"sht":[[0,0,0],[0,0,0],[0,0,0],[0,0,0]],"ver":1} (retained)
17:58:00.445 MQT: tasmota/discovery/4C75250C1932/sensors = {"sn":{"Time":"2023-06-25T17:58:00","SR04":{"Distance":0.2}},"ver":1} (retained)
17:58:05.432 MQT: tele/tasmota_0C1932/STATE = {"Time":"2023-06-25T17:58:05","Uptime":"0T00:00:20","UptimeSec":20,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"11n","RSSI":98,"Signal":-51,"LinkCount":1,"Downtime":"0T00:00:05"}}
17:58:05.441 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:58:05","SR04":{"Distance":0.2}}
17:59:15.442 MQT: tele/tasmota_0C1932/SENSOR = {"Time":"2023-06-25T17:59:15","SR04":{"Distance":0.2}}

**Comment by Jaegerstefan** [comment]
I face the same issue with a JSN-SR04T here on a esp32 version. Sensor is detected in mode 1, but no reasonable values. Tested with two JSN-SR04T, so i dont think it's related to the sensor itself. 
Powered by separate 5V supply or through the ESP32 dev boards

**Reply by Jaegerstefan** [reply]
By the way, the same sensor / board works with esphome successfully.
